sirepo/package_data/template/radia/geometry.py.jinja

```diff
@@ -58,35 +57,58 @@ def _add_object(o):
         radia_objs.append(g_id)
     return g_id
 
-id_map = PKDict()
+def _build_radia_geom():
+    id_map = PKDict()
+    radia_objs = []
 {% if isRaw %}
-g_id, id_map = radia_examples.EXAMPLES['{{ exampleName }}']()
+    g_id, id_map = radia_examples.EXAMPLES['{{ exampleName }}']()
 {% elif dmpImportFile %}
-with open('{{ dmpImportFile }}', 'rb') as f:
-    b = f.read()
-    g_id = radia_tk.load_bin(b)
+    with open('{{ dmpImportFile }}', 'rb') as f:
+        b = f.read()
+        g_id = radia_tk.load_bin(b)
 {% else %}
-for obj in sirepo_objs:
-    o = PKDict(obj)
-    if o.model not in _MODELS or o.id in id_map.values():
-        continue
-    o_id = _add_object(o)
-    id_map[o_id] = o.id
-# put all objects in a container for now
-g_id = radia_tk.build_container(radia_objs)
-#id_map[g_id] = '{{ geomId }}'
+    for obj in sirepo_objs:
+        o = PKDict(obj)
+        if o.model not in _MODELS or o.id in id_map.values():
+            continue
+        o_id = _add_object(o, radia_objs, id_map)
+        id_map[o_id] = o.id
+    g_id = radia_tk.build_container(radia_objs)
+{% endif %}
+    # free symmetries to test parallel?
+    #g_id = radia_tk.free_symmetries(g_id)
+    return g_id, id_map
+
+
+def _write_h5_data(data, h5_path):
+    with h5py.File('{{ dataFile }}', 'a') as hf:
+        template_common.dict_to_h5(data, hf, path=h5_path)
```

Comment:
c1f0ea7

Incomplete, because I'm not sure what to do with what is now ```write_dict_to_h5()```.  Is ```with h5py.File``` to move there?  If so, is it better to have once at the top or every time we call ```create_dataset```?  Any issues for a recursive function?

Reply:
918c953 7a01b53 0c28245

 I answered my own question: ```with h5py.File()``` can live in ```write_dict_to_h5()``` as long it's done only when actually creating the dataset.  Some other points:

- The only access mode that makes sense is 'a' so I moved that in as well
- ```kickMap.py.jinja``` is not currently used but probably will be since it should be susceptible to parallel processing.  I  updated it.  The ```_write_dict_to_h5()``` there is repeated but can be moved to a separate jinja file when ready
- warpvnd uses ```template_common.write_dict_to_h5()``` in one place only, so it has been updated.  Its jinja files use a local version and I am not going to touch those for now
- I am also not going to change the corresponding ```h5_to_dict()``` yet since other codes use it and this PR needs to get folded in.  Should make a separate issue for it